modules/ai/assets/js/editor/components/prompt-history/context/prompt-history-context.js

```diff
@@ -0,0 +1,133 @@
+import { createContext, useContext, useEffect, useState } from 'react';
+import { HISTORY_TYPES } from '../history-types';
+import { vh } from '../../../helpers';
+
+export const PromptHistoryContext = createContext( {} );
+
+const DIALOG_CONTAINER_SELECTOR = '.e-ai-dialog-content';
+
+const getContainer = () => {
+	return document.querySelector( DIALOG_CONTAINER_SELECTOR );
+};
+
+const getContainerHeight = () => {
+	const contentContainer = document.querySelector( DIALOG_CONTAINER_SELECTOR );
+	return contentContainer?.clientHeight;
+};
+
+const getContentHeight = () => {
+	const contentContainer = document.querySelector( DIALOG_CONTAINER_SELECTOR );
+	return contentContainer.children[ 0 ].clientHeight + 50;
+};
+
+const setContentHeight = ( height ) => {
+	const contentContainer = document.querySelector( DIALOG_CONTAINER_SELECTOR );
+
+	if ( 'auto' === height ) {
+		return contentContainer.style.height = 'auto';
+	}
+
+	contentContainer.style.height = `${ height }px`;
+};
+
+export const PromptHistoryProvider = ( { promptType, children } ) => {
+	const showHistoryState = useState( false );
+
+	return (
+		<PromptHistoryContext.Provider value={ { promptType, showHistoryState } }>
+			{ children }
+		</PromptHistoryContext.Provider>
+	);
+};
+
+PromptHistoryProvider.propTypes = {
+	promptType: PropTypes.oneOf( Object.values( HISTORY_TYPES ) ).isRequired,
+	children: PropTypes.node,
+};
+
+export const usePromptHistoryContext = () => {
+	const { promptType, showHistoryState } = useContext( PromptHistoryContext );
+	const [ showHistory, setShowHistory ] = showHistoryState;
+
+	const [ showModalWrapper, setShowModalWrapper ] = useState( false );
+	const [ showPromptHistory, setShowPromptHistory ] = useState( false );
+
+	const handleFullScreenHistoryState = () => {
+		if ( showHistory ) {
+			setShowModalWrapper( true );
+			setTimeout( () => setShowPromptHistory( true ), 200 );
+
+			return;
+		}
+
+		setShowPromptHistory( false );
+		setTimeout( () => setShowModalWrapper( false ), 500 );
```

Comment:
why not pass the `showHistory` in to `setShowModalWrapper` method.
Then you can extract it from the if?
setShowModalWrapper( showHistory );

Reply:
Can you explain a little more? 🙂 

Bc now I don't see how we can extract it as if `showHistory` is `true` then we have to change the value immediately, but if it's `false` -- we have to do it on timeout.

I think we can change booleans with the `showHistory` reference, but would that be helpful? 🤔

Reply:
```
setShowModalWrapper( showHistory );
		if ( showHistory ) {
			setTimeout( () => setShowPromptHistory( true ), 200 );
			return;
		}
		setTimeout( () => setShowModalWrapper( false ), 500 );
```
		
		Like so.

Reply:
So if `showHistory === false`, it will be called twice:

1. `setShowModalWrapper( showHistory );`
2. `setTimeout( () => setShowModalWrapper( false ), 500 );`

And that will break the closing animation.

Why would we want to do that? 🤔